tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -103,8 +41,14 @@ private static void RecordTelemetry<T>(IConfigurationTelemetry telemetry, string
             case bool boolVal:
                 telemetry.Record(key, boolVal, ConfigurationOrigins.Default);
                 break;
+            case string stringVal:
+                telemetry.Record(key, stringVal, recordValue, ConfigurationOrigins.Default);
+                break;
+            case null: // can't actually be called in practice
+                break;
             default:
-                telemetry.Record(key, defaultValue.TelemetryValue, recordValue, ConfigurationOrigins.Default);
+                // TODO: this shouldn't be calleable in practice, we need to revise it
```

Comment:
Would require a bunch more refactoring and copy-pasting, so will handle in a follow up (if at all)